addons/binding/org.openhab.binding.unifi/src/main/java/org/openhab/binding/unifi/internal/handler/UniFiClientThingHandler.java

```diff
@@ -111,14 +114,30 @@ public void initialize() {
 
     @Override
     public void handleCommand(ChannelUID channelUID, Command command) {
+        logger.debug("Handling command = {} for channel = {}", command, channelUID);
+
         // mgb: only handle the command if we're ONLINE
         if (getThing().getStatus() == ONLINE) {
-            logger.debug("Handling command = {} for channel = {}", command, channelUID);
-            if (command == REFRESH) {
+
+            if (command instanceof OnOffType && channelUID.getId().equals(CHANNEL_BLOCKED)) {
+                setBlockStateChannel(getClient(), channelUID, command);
+            } else if (command == REFRESH) {
                 refreshChannel(getClient(), channelUID);
             } else {
-                logger.debug("Ignoring unsupported command = {} for channel = {} - the UniFi binding is read-only!",
-                        command, channelUID);
+                logger.debug("Ignoring unsupported command = {} for channel = {}!", command, channelUID);
+            }
+        }
+    }
+
+    private void setBlockStateChannel(@Nullable UniFiClient client, ChannelUID channelUID, Command command) {
+        logger.debug("BLOCK state to be set: {} for client {}", command, client.getId());
+
+        if (client != null && controllerHandler != null && controllerHandler.getController() != null) {
+            try {
+                String siteName = controllerHandler.getSiteNameForClient(client);
```

Comment:
Not sure if you need to search for it. The sitename is also available in `config.getSite()`. This is what is used to in `getClient`. I would assume that should also work here.

Reply:
I think the config.getSite() is from configuration made in OH (which is optional). I wanted to get site name (not ID) from "running" state.

Reply:
I haven't made any changes here yet. Do you think the extra API calls for every refresh would be OK? Cause I think we only get "site ID" on the "/stat/sta"-call, which will result in additional calls to get the Site for this ID (which then will have Site Name included).